### PR TITLE
improvements to eval

### PR DESCRIPTION
**Pull request by hturki**
- Handle case when N_importance is 0
- Generate video for depth maps as well

### CODE REVIEW

File: eval.py [hunk 1]
```diff
@@ -58,8 +56,7 @@ def get_opts():
 @torch.no_grad()
 def batched_inference(models, embeddings,
                       rays, N_samples, N_importance, use_disp,
-                      chunk,
-                      white_back):
```
Comment (kwea123) [inline eval.py:62]: `dataset` is not visible inside this function, so `white_back` in the argument is required.

File: eval.py [hunk 2]
```diff
@@ -117,14 +117,15 @@ def batched_inference(models, embeddings,
         rays = sample['rays'].cuda()
         results = batched_inference(models, embeddings, rays,
                                     args.N_samples, args.N_importance, args.use_disp,
-                                    args.chunk,
-                                    dataset.white_back)
```
Comment (kwea123) [inline eval.py:121]: again, passing in `dataset.white_back` here is required

File: eval.py [hunk 3]
```diff
@@ -117,14 +117,15 @@ def batched_inference(models, embeddings,
...
+
+        img_pred = results[f'rgb_{typ}'].view(h, w, 3).cpu().numpy()
 
-        img_pred = results['rgb_fine'].view(h, w, 3).cpu().numpy()
-        
         if args.save_depth:
-            depth_pred = results['depth_fine'].view(h, w).cpu().numpy()
-            depth_pred = np.nan_to_num(depth_pred)
+            depth_pred = results[f'depth_{typ}'].view(h, w).cpu().numpy()
+            depth_pred_ = (depth_pred * 255).astype(np.uint8)
```
Comment (kwea123) [inline eval.py:127]: Have you verified that this generates visually-plausible (and consistent) depth images? From my first sight no, because `depth_pred` is depth value, it can be anything like 1 or 10 or 3.1415926. Simply multiplying by 255 will not do the trick. You can refer to my `utils/visualization.py` to see how I draw depth images in the tensorboard log, but I'm not sure this generates visually-consistent depth images across the whole video (probably not). You need to sort of fix a minimum and maximum depth value across the whole video and normalize every depth according to them, then finally draw the depth images.

**Review by kwea123 — CHANGES_REQUESTED** [review]
The depth image part is tricky and probably requires fix (a depth video showing that your code works would be great).

The evaluation without fine model is generally good and only requires minor fixes.

Please see my inline comments. Thank you for the contribution.

File: eval.py (hunk 1 above)
Comment (hturki) [reply eval.py:62]: so it's unintuitive, but if you look at the code below, you're never actually calling white_back but instead dataset.white_back (I discovered this because my IDE was flagging the white_back parameter as unused)

The scoping here is really unintuitive here but the IDE and running the code suggest that dataset.white_back is resolving properly in the function)

File: eval.py (hunk 3 above)
Comment (hturki) [reply eval.py:127]: good point, will fix

**Comment by hturki** [comment]
Here's what things look like with the latest commit:

rgb video (using only coarse nerf):

test

associated depth video:

test_depth

**Review by kwea123 — APPROVED** [review]
Looks good to me now. Yes you're correct, white_back was not used in the function.

**Comment by kwea123** [comment]
btw, this is what I get after fixing the data:
train
train_depth

**Comment by kwea123** [comment]
Also I added a `--split test_train` split that allows evaluation on training images. It is useful when the training images are already dense and roughly follows a spherical path.
train2
train2_depth
